ProcessLib/SmallDeformationWithLIE/LocalAssembler/SmallDeformationLocalAssemblerMatrixNearFracture-impl.h

```diff
@@ -163,30 +209,36 @@ assembleWithJacobian(
         auto& C = ip_data._C;
         auto& material_state_variables = *ip_data._material_state_variables;
 
-        eps.noalias() = B * nodal_u;
+        eps.noalias() = B * nodal_total_u;
 
         if (!ip_data._solid_material.computeConstitutiveRelation(
                 t, x_position, _process_data.dt, eps_prev, eps, sigma_prev,
                 sigma, C, material_state_variables))
             OGS_FATAL("Computation of local constitutive relation failed.");
 
-        // r_ru = B^T Stress = B^T C B (u + phi*[u])
-        localRhs_ru.noalias() -= B.transpose() * sigma * detJ * wp.getWeight() * integralMeasure;
-
-        // r_[u] = (phi*B)^T Stress = (phi*B)^T C B (u + phi*[u])
-        localRhs_du.noalias() -= levelsets * B.transpose() * sigma * detJ * wp.getWeight() * integralMeasure;
+        // r_u = B^T * Sigma = B^T * C * B * (u+phi*[u])
+        // r_[u] = (phi*B)^T * Sigma = (phi*B)^T * C * B * (u+phi*[u])
+        local_b_u.noalias() -= B.transpose() * sigma * ip_factor;
+        for (unsigned i=0; i<n_fractures; i++)
+            vec_local_b_g[i].noalias() -= levelsets[i] * B.transpose() * sigma * ip_factor;
```

Comment:
Is it really better to store the Eigen::Blocks in a std:vector instead of building them on demand? E.g., `vec_local_b_g` seems to be used only here.